conda/common/path.py

```diff
@@ -330,7 +331,10 @@ def _translation(found_path):  # NOQA
 
 
 def which(executable):
-    return find_executable(executable)
+    """Backwards-compatibility wrapper. Use `shutil.which` directly if possible."""
+    from shutil import which
+
+    return which(executable)
```

Comment:
This should be marked for deprecation if there is a stdlib alternative

Reply:
Yes, but we need to update all `which` uses in the project first. I didn't do it yet so this PR doesn't grow too much. My concern is that if we add a DeprecationWarning and some config tells Python to print these (e.g. env var) this might pollute the output in some commands. Although it should go to stderr so maybe not that big of a deal.

Reply:
I agree that it's useful to redirect calls through an alias to be able to change the way conda works with one edit, instead of many; an alias has no maintenance cost; and there are other more profitable deprecations such as any code that is right now in the way of producing a user-visible improvement to the product.

Reply:
@dholth This makes it sound like you're not in favor of our deprecation policy, which I beg to differ. Deprecation is needed (and it's totally fine to do it when the actual uses of which are changed) since many 3rd party libraries are importing random stuff from conda. The deprecations are the way to tell them about it.